usaspending_api/awards/v2/filters/location_filter_geocode.py

```diff
@@ -108,20 +128,71 @@ def get_fields_list(scope, field_value):
     return [field_value]
 
 
-def return_query_string(use_matview):
-    # Returns query strings according based on mat view or database
+def return_query_string(use_matview: bool) -> tuple:
+    # Returns query strings based upon if the queryset is for a normalized or de-normalized model
 
-    if use_matview:
+    if use_matview:  # de-normalized
         # Queries going through the references_location table will not require a join
         # Example "pop__county_code"
         q_str = '{0}_{1}'
-        # Matviews use country_code ex: pop_country_code
-        country_code_col = 'country_code'
+        country_code_col = 'country_code'  # Matviews use country_code ex: pop_country_code
     else:
-        # Queries going through the references_location table will a join in the filter
+        # Queries going through the references_location table will require a join in the filter
         # Example "place_of_performance__county_code"
         q_str = '{0}__{1}'
-        # References_location table uses the col location_country_code
-        country_code_col = 'location_country_code'
+        country_code_col = 'location_country_code'  # References_location table uses the col location_country_code
 
     return q_str, country_code_col
+
+
+def create_city_name_queryset(scope: str, list_of_cities: list, country_code: str, state_code: Optional[str] = None):
+    """
+        Given a list of city names and the scope, return a django queryset.
+        scope = "pop" or "recipient_location"
+        list_of_city_names is a list of strings
+        country_code is the country code to limit the query and results from Elasticsearch
+        state_code (optional) is the state code if the search should be limited to that state
+    """
+    matching_awards = set(
+        chain(*[get_award_ids_by_city(scope, city, country_code, state_code) for city in list_of_cities])
+    )
+    result_queryset = Q(pk=None)  # If there are no city results in Elasticsearch, use this always falsey Q filter
+
+    if matching_awards:
+        result_queryset = Q(**{"award_id" + "__in": matching_awards})
+    return result_queryset
+
+
+def get_award_ids_by_city(scope: str, city: str, country_code: str, state_code: Optional[str] = None) -> list:
+    """
+    """
+    # Search using a "filter" instead of a "query" to leverage ES caching
+    query = {
+        "bool": {
+            "must": [
+                {"match": {"{}_city_name".format(scope): es_sanitize(city)}},
```

Comment:
Instead of running multiple `es_sanitize`s here, would it make more sense to add a boolean flag `sanitize` for `elasticsearch_results` and sanitize the whole body there?

Reply:
To make that work technically would require modifying the sanitize function to parse json, but if we accomplish that I would suggest putting it right in the client so that we don't have to worry about not sanitizing an endpoint by accident.

Reply:
I think it makes sense to put it in the client but that's just my take. I just wanted to prevent multiple sanitization calls when it can sanitize everything before it's processed.

Reply:
How do you feel about the changes proposed here? https://github.com/fedspendingtransparency/usaspending-api/pull/1832/files
I haven't been able to test them yet or run them by the rest of API team

Reply:
@dpb-bah We can't sanitize the entire JSON because we would be stripping out characters necessary to complete the searches.